agents/loco_mc_agent.py

```diff
@@ -56,6 +56,10 @@ def __init__(self, opts, name=None):
         if opts.log_timeline:
             self.timeline_log_file = open("timeline_log.{}.txt".format(self.name), "a+")
 
+        # Add optional hook for perceive
+        if opts.enable_timeline:
+            self.memory.register_hook(self.log_to_dashboard, self.perceive)
```

Comment:
why is `self.memory` registering a hook for `self.perceive`? this makes no sense.

Does this ever get triggered at all?

Reply:
Thanks for pointing this out! It was being triggered previously, so now I've modified `register_hook` in memory to verify senders and added a new `register_hook` to loco_mc_agent.py.